### ISSUE TITLE
Value List Item child buttons - Provide a valid label for form fields - (2036570432)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5622]
### Module: 
 39 Value List and Value List Item 

### Violation: 
 Provide a valid label for form fields 

![image](https://user-images.githubusercontent.com/5023024/202569516-7f3a7729-ff6c-4efa-98ef-5c2d8ca27701.png)

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570432 

### Severity: 
 10 

### Description: 
 [Issue]
There are remove buttons after "Test 4. Value List with Non-interactive and Removable List Items" text with inappropriate and identical accessible names.

[User Impact]
Screen reader users will be unable to determine the purpose of these buttons. Speech input users will have difficulty activating them.

[Code Reference] 
 ```
<button aria-busy="false" aria-disabled="false" aria-label="Remove" class="button">
  (...)
</button>
 ``` 

### Note: 
 [Suggestion]
Provide a valid label for form fields. This includes providing accessible names for buttons. The accessible name of a button can be set with internal text, an aria-label attribute, or an aria-labelledby attribute. Good accessible names are both concise and descriptive.

[Compliant Code Example] 
 ```
<button aria-busy="false" aria-disabled="false" aria-label="Remove river" class="button">
  (...)
</button>
 ```
 

### Media Type: 
 Forms 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Similar to #5623, may be out of scope with the updated `list` component, as this should be handled with the updated refactor.

Leaving open for confirmation, but likely can be closed once confirmed.

**Comment by driskull** [comment]
This could be fixed by the end user.
